### PR TITLE
chore: validate Dockerfile parsing against Docker's parser

### PR DESCRIPTION
**Pull request by efekarakus**
This PR contains some small refactors, but more importantly adds additional unit tests to ensure that our `EXPOSE` and `HEALTHCHECK` parsers match Docker's.

Once this change is merged, a follow-up will remove Docker as a dependency from the CLI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/docker/dockerfile/dockerfile_test.go [hunk 1]
```diff
@@ -152,8 +81,11 @@ EXPOSE 80`),
 		},
 		"bad expose token single port": {
 			dockerfilePath: wantedPath,
-			dockerfile:     []byte(`EXPOSE $arg`),
-			wantedPorts: []portConfig{
+			dockerfile: []byte(`
+FROM nginx
+EXPOSE $arg
+`),
+			wantedPorts: []Port{
```
Comment (Lou1415926) [inline internal/pkg/docker/dockerfile/dockerfile_test.go]: nit: seems like we'd wanted `nil` for ports

**Review by Lou1415926 — APPROVED** [review]
lgtm! just one nit on a unit test!

File: internal/pkg/docker/dockerfile/dockerfile.go [hunk 2]
```diff
@@ -62,50 +70,59 @@ type HealthCheck struct {
...
+// New returns an empty Dockerfile.
+func New(fs afero.Fs, path string) *Dockerfile {
 	return &Dockerfile{
-		ExposedPorts: []portConfig{},
-		HealthCheck:  nil,
+		exposedPorts: []Port{},
+		healthCheck:  nil,
 		fs:           fs,
 		path:         path,
 		parsed:       false,
```
Comment (iamhopaul123) [inline internal/pkg/docker/dockerfile/dockerfile.go]: Probably can be just
```suggestion
		fs:           fs,
		path:         path,
```

**Review by iamhopaul123 — APPROVED** [review]
🚢

File: internal/pkg/docker/dockerfile/dockerfile_test.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/docker/dockerfile/dockerfile_test.go]: I'm a dummy yeah! I changed this to `nil` and added a `require.Nil` (although it wouldn't have caught this)

File: internal/pkg/docker/dockerfile/dockerfile.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/docker/dockerfile/dockerfile.go]: Done!

**Pull request merged**